src/system/SystemTimer.cpp

```diff
@@ -160,5 +160,23 @@ TimerList TimerList::ExtractEarlier(Clock::Timestamp t)
     return out;
 }
 
+Clock::Timestamp TimerList::GetRemainingTime(TimerCompleteCallback aOnComplete, void * aAppState)
+{
+    for (TimerList::Node * timer = mEarliestTimer; timer != nullptr; timer = timer->mNextTimer)
+    {
+        if (timer->GetCallback().GetOnComplete() == aOnComplete && timer->GetCallback().GetAppState() == aAppState)
+        {
+            Clock::Timestamp currentTime = SystemClock().GetMonotonicTimestamp();
+
+            if (currentTime < timer->AwakenTime())
+            {
+                return timer->AwakenTime() - currentTime;
+            }
+            return Clock::kZero;
```

Comment:
Given this implementation, IsTimerActive will return false for timers that are not-yet-fired but whose time has expired while we were running this code.

I strongly suggest pulling the timer changes out into a separate PR that:

1) Will be able to get careful reviews.
2) Will have unit tests covering all the exciting edge cases here.

Reply:
In the case of the ICD usage. Timer expired will also allow me  to extend the active time by restarting the timer for an activeThreshold interval which. That work for my use case, but I guess in the general sense of the extendTimeraTo and IsActive it would be better to make a distinction.

Reply:
>  So the new separate PR might get merged before.

That sounds perfect, right?

Reply:
Yes I agree